src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1907,30 +1911,54 @@ private static IEnumerable<InstrumentationKind> ParseInstrumentationKinds(string
             return new ResourceDescription(resourceName, fileName, dataProvider, isPublic, embedded, checkArgs: false);
         }
 
-        private static IEnumerable<string> ParseWarnings(string value)
+        private static string ParseWarning(string id)
+        {
+            if (ushort.TryParse(id, NumberStyles.Integer, CultureInfo.InvariantCulture, out ushort number) &&
+                       ErrorFacts.IsWarning((ErrorCode)number))
+            {
+                // The id refers to a compiler warning.
+                return CSharp.MessageProvider.Instance.GetIdForErrorCode(number);
+            }
+            else
+            {
+                // Previous versions of the compiler used to report a warning (CS1691)
+                // whenever an unrecognized warning code was supplied in /nowarn or 
+                // /warnaserror. We no longer generate a warning in such cases.
+                // Instead we assume that the unrecognized id refers to a custom diagnostic.
+                return id;
+            }
+        }
+
+        private static IEnumerable<string> ParseNoWarnWarnings(string value)
         {
             value = value.Unquote();
             string[] values = value.Split(new char[] { ',', ';', ' ' }, StringSplitOptions.RemoveEmptyEntries);
             foreach (string id in values)
             {
-                ushort number;
-                if (ushort.TryParse(id, NumberStyles.Integer, CultureInfo.InvariantCulture, out number) &&
-                    ErrorFacts.IsWarning((ErrorCode)number))
+                if (id.ToLowerInvariant() == "nullable")
                 {
-                    // The id refers to a compiler warning.
-                    yield return CSharp.MessageProvider.Instance.GetIdForErrorCode(number);
+                    foreach (var errorCode in ErrorFacts.NullableFlowAnalysisWarnings)
+                    {
+                        yield return errorCode;
+                    }
                 }
                 else
                 {
-                    // Previous versions of the compiler used to report a warning (CS1691)
-                    // whenever an unrecognized warning code was supplied in /nowarn or 
-                    // /warnaserror. We no longer generate a warning in such cases.
-                    // Instead we assume that the unrecognized id refers to a custom diagnostic.
-                    yield return id;
+                    yield return ParseWarning(id);
                 }
             }
         }
 
+        private static IEnumerable<string> ParseWarnAsErrorWarnings(string value)
```

Comment:
>ParseWarnAsErrorWarnings [](start = 43, length = 24)

This raises an interesting question: should we allow `warnAsError:nullable` the same way we allow `noWarn:nullable`.
@alekseyts, @gafter, @cston, what do you think?

Reply:
Aleksey also thought supporting `warnAsError:nullable` would be good and consistent.
Could you factor the code back together?

---
In reply to: [294552006](https://github.com/dotnet/roslyn/pull/36464#discussion_r294552006) [](ancestors = 294552006)

Reply:
re-activated this

---
In reply to: [295037412](https://github.com/dotnet/roslyn/pull/36464#discussion_r295037412) [](ancestors = 295037412,294552006)